### ISSUE TITLE
mpg123-sdl-1.31.1-1: error: implicit declaration of function 'memset_pattern4' is invalid in C99

### ISSUE THREAD

**Issue by cooljeanius** [fink/fink-distributions#1099]
Another victim of `-Werror=implicit-function-declaration` being on by default now:
```
make src/libout123/modules/output_sdl.la
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I./src  -DPKGLIBDIR="\"/opt/sw/lib/mpg123\""  -I./src -I./src/compat -I./src/libmpg123 -I./src/libout123 -I./src/libmpg123 -I./src/libsyn123 -I./src/libout123 -DOPT_MULTI -DOPT_X86_64 -DOPT_GENERIC -DOPT_GENERIC_DITHER -DREAL_IS_FLOAT -DOPT_AVX -I/opt/sw/include -shared -I/opt/sw/include -I/opt/sw/include/SDL2 -D_THREAD_SAFE -O3 -fomit-frame-pointer -funroll-all-loops -finline-functions -ffast-math  -g -c -o src/libout123/modules/output_sdl_la-sdl.lo `test -f 'src/libout123/modules/sdl.c' || echo './'`src/libout123/modules/sdl.c
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I./src  -DPKGLIBDIR="\"/opt/sw/lib/mpg123\""  -I./src -I./src/compat -I./src/libmpg123 -I./src/libout123 -I./src/libmpg123 -I./src/libsyn123 -I./src/libout123 -DOPT_MULTI -DOPT_X86_64 -DOPT_GENERIC -DOPT_GENERIC_DITHER -DREAL_IS_FLOAT -DOPT_AVX -I/opt/sw/include  -O3 -fomit-frame-pointer -funroll-all-loops -finline-functions -ffast-math  -g -c -o src/compat/compat_str.lo src/compat/compat_str.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./src -DPKGLIBDIR=\"/opt/sw/lib/mpg123\" -I./src -I./src/compat -I./src/libmpg123 -I./src/libout123 -I./src/libmpg123 -I./src/libsyn123 -I./src/libout123 -DOPT_MULTI -DOPT_X86_64 -DOPT_GENERIC -DOPT_GENERIC_DITHER -DREAL_IS_FLOAT -DOPT_AVX -I/opt/sw/include -I/opt/sw/include -I/opt/sw/include/SDL2 -D_THREAD_SAFE -O3 -fomit-frame-pointer -funroll-all-loops -finline-functions -ffast-math -g -c src/libout123/modules/sdl.c  -fno-common -DPIC -o src/libout123/modules/.libs/output_sdl_la-sdl.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./src -DPKGLIBDIR=\"/opt/sw/lib/mpg123\" -I./src -I./src/compat -I./src/libmpg123 -I./src/libout123 -I./src/libmpg123 -I./src/libsyn123 -I./src/libout123 -DOPT_MULTI -DOPT_X86_64 -DOPT_GENERIC -DOPT_GENERIC_DITHER -DREAL_IS_FLOAT -DOPT_AVX -I/opt/sw/include -O3 -fomit-frame-pointer -funroll-all-loops -finline-functions -ffast-math -g -c src/compat/compat_str.c  -fno-common -DPIC -o src/compat/.libs/compat_str.o
clang: warning: optimization flag '-funroll-all-loops' is not supported [-Wignored-optimization-argument]
clang: warning: optimization flag '-funroll-all-loops' is not supported [-Wignored-optimization-argument]
In file included from src/libout123/modules/sdl.c:16:
In file included from /opt/sw/include/SDL2/SDL.h:32:
In file included from /opt/sw/include/SDL2/SDL_main.h:25:
/opt/sw/include/SDL2/SDL_stdinc.h:532:5: error: implicit declaration of function 'memset_pattern4' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    memset_pattern4(dst, &val, dwords * 4);
    ^
/bin/sh ./libtool  --tag=CC   --mode=link gcc  -O3 -fomit-frame-pointer -funroll-all-loops -finline-functions -ffast-math  -g  -L/opt/sw/lib -Wl,-headerpad_max_install_names -o src/compat/libcompat_str.la  src/compat/compat_str.lo  
1 error generated.
make: *** [Makefile:4663: src/libout123/modules/output_sdl_la-sdl.lo] Error 1
make: *** Waiting for unfinished jobs....
libtool: link: ar cr src/compat/.libs/libcompat_str.a src/compat/.libs/compat_str.o 
libtool: link: ranlib src/compat/.libs/libcompat_str.a
libtool: link: ( cd "src/compat/.libs" && rm -f "libcompat_str.la" && ln -s "../libcompat_str.la" "libcompat_str.la" )
### execution of make failed, exit code 2
### execution of /tmp/fink.iY_gw failed, exit code 2
Removing runtime build-lock...
Removing build-lock package...
/opt/sw/bin/dpkg-lockwait -r fink-buildlock-mpg123-sdl-1.31.1-1
(Reading database ... 
dpkg: serious warning: files list file for package `fink-buildlock-ccp4-6.5-1' missing, assuming package has no files currently installed.
1037365 files and directories currently installed.)
Removing fink-buildlock-mpg123-sdl-1.31.1-1 ...
```
(I'm on x86_64 Big Sur with Xcode 13)
cc @danielj7

**Comment by dmacks** [comment]
fixed in b05c4935d082c93d200d67767d6197131289382c

**Comment by cooljeanius** [comment]
> fixed in [b05c493](https://github.com/fink/fink-distributions/commit/b05c4935d082c93d200d67767d6197131289382c)

confirmed; thanks!
